src/components/typings/OptionGroup.ts

```diff
@@ -1,11 +1,11 @@
 import * as React from "react";
+import { SearchProps } from "./Search";
 
 export interface OptionGroupProps_ {
   children: React.ReactNodeArray;
   className?: string;
   searchBox?: boolean;
-  onSearchBoxQueryChange?: (query: string) => void;
-  searchBoxPlaceholder?: string;
+  searchBoxProps?: Partial<SearchProps>;
```

Comment:
so I don't know how feasible and maintainable two props dependent on one another.
```ts
export type OptionGroupProps_ = {
  className?: string;
  ... // others
} & {
  searchBox: true;
  searchBoxProps: SearchProps
}
```
Such that it determines that if searchBox is true, searchBoxProps is defined and contains required props of SearchProps.

Don't do this now as there is some bug as it will ask you to add default props in searchBoxProps even though React adds them on its own. Just keep this in mind when you need two keys dependent on one another.